src/mpyl/steps/deploy/dagster.py

```diff
@@ -0,0 +1,180 @@
+                Constants.DAGSTER_WORKSPACE_CONFIGMAP,
+            )
+            dagster_workspace = yaml.safe_load(
+                config_map.data[Constants.DAGSTER_WORKSPACE_FILE]
+            )
+
+            server_names = [
+                w["grpc_server"]["location_name"]
+                for w in dagster_workspace["load_from"]
+            ]
+
+            # If the server new (not in existing workspace.yml), we append it
+            if user_code_name_to_deploy not in server_names:
+                self._logger.info(
+                    f"Adding new server {user_code_name_to_deploy} to dagster's workspace.yaml"
+                )
+                dagster_workspace["load_from"].append(
+                    to_grpc_server_entry(
+                        host=user_code_deployment["deployments"][0]["name"],
+                        port=user_code_deployment["deployments"][0]["port"],
+                        location_name=user_code_deployment["deployments"][0]["name"],
+                    )
+                )
+                updated_config_map = update_config_map_field(
+                    config_map=config_map,
+                    field=Constants.DAGSTER_WORKSPACE_FILE,
+                    data=dagster_workspace,
+                )
+                configmap_update_result = replace_config_map(
+                    core_api,
+                    Constants.DAGSTER_NAMESPACE,
+                    Constants.DAGSTER_WORKSPACE_CONFIGMAP,
+                    updated_config_map,
+                )
+                if configmap_update_result.success:
+                    self._logger.info(
+                        f"Successfully added {user_code_name_to_deploy} to dagster's workspace.yaml"
+                    )
+                    dagster_deploy_results.append(configmap_update_result)
+                if not configmap_update_result.success:
+                    return configmap_update_result
+
+            # restarting ui and daemon
+            rollout_restart_output = rollout_restart_deployment(
+                self._logger,
+                apps_api,
+                Constants.DAGSTER_NAMESPACE,
+                Constants.DAGSTER_DAEMON,
+            )
+            if rollout_restart_output.success:
+                self._logger.info(rollout_restart_output.message)
+                dagster_deploy_results.append(rollout_restart_output)
+                rollout_restart_output = rollout_restart_deployment(
+                    self._logger,
+                    apps_api,
+                    Constants.DAGSTER_NAMESPACE,
+                    Constants.DAGSTER_DAGIT,
+                )
+                if not rollout_restart_output.success:
+                    return rollout_restart_output
+
+                self._logger.info(rollout_restart_output.message)
+                dagster_deploy_results.append(rollout_restart_output)
+            else:
+                return rollout_restart_output
+
+        return (
```

Comment:
Shouldn't you remove the other returns (lines 170, 165, 146) and add them to `dagster_deploy_results` instead, in order to fully utilize `__flatten_result_messages`?

Reply:
[d88a106](https://github.com/Vandebron/mpyl/pull/146/commits/d88a1066332f567615ea24bfb55f17842cf79cfa)

addressed this here

Reply:
looks way nicer with processing your comment!